pro/src/data_access/ApiRecipeDataAccessObject.java

```diff
@@ -1,68 +1,83 @@
 package data_access;
 
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
 import okhttp3.OkHttpClient;
 import okhttp3.Request;
 import okhttp3.Response;
 import org.json.JSONObject;
+import service.search.use_case.SearchDataAccessInterface;
 
-import java.io.IOException;
-import java.util.HashMap;
-
-
-public class ApiRecipeDataAccessObject {
+public class ApiRecipeDataAccessObject implements SearchDataAccessInterface {
     private final String keyword;
 
-    public ApiRecipeDataAccessObject(String keyword){
+    public ApiRecipeDataAccessObject(String keyword) {
         this.keyword = keyword;
     }
 
-    public JSONObject content(String keyword) throws IOException {
+    private JSONObject content(String keyword) throws IOException {
         OkHttpClient client = new OkHttpClient();
-
         String newKeyword = keyword.replace(" ", "_");
-        String link ="https://www.themealdb.com/api/json/v1/1/search.php?lat=%7Blat%7D&s=" + newKeyword;
-
-        Request request = new Request.Builder()
-                .url(link)
-                .get()
-                .build();
-
+        String link = "https://www.themealdb.com/api/json/v1/1/search.php?lat=%7Blat%7D&s=" + newKeyword;
+        Request request = (new Request.Builder()).url(link).get().build();
         Response response = client.newCall(request).execute();
         JSONObject responseBody = new JSONObject(response.body().string());
         return responseBody;
     }
 
-    public HashMap<String, Object> filter(JSONObject recipe_json, int index){
+    private HashMap<String, Object> filter(JSONObject recipe_json, int index) {
```

Comment:
The method needs to return Recipe, which is an entity, an instance of the interface Recipe.